src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -759,22 +759,27 @@ export default class ResultsViewOncoprint extends React.Component<IResultsViewOn
         invoke:()=>{
             const groupedSelectedMolecularProfiles:{[alterationType:string]:MolecularProfile[]} =
                 _.groupBy(this.props.store.selectedMolecularProfiles.result!, "molecularAlterationType");
+            const selectedMolecularProfilesMap = _.keyBy(this.props.store.selectedMolecularProfiles.result!, p=>p.molecularProfileId);
 
             const existsUnprofiled:{[alterationType:string]:boolean} = {};
             const coverageInfo = this.props.store.coverageInformation.result!.samples;
             const molecularProfileIdToMolecularProfile = this.props.store.molecularProfileIdToMolecularProfile.result!;
             for (const uniqueSampleKey of Object.keys(coverageInfo)) {
                 for (const gpData of coverageInfo[uniqueSampleKey].notProfiledAllGenes) {
-                    existsUnprofiled[
-                        molecularProfileIdToMolecularProfile[gpData.molecularProfileId].molecularAlterationType
-                    ] = true;
+                    if (gpData.molecularProfileId in selectedMolecularProfilesMap) {
```

Comment:
for the loops here, i'm in favor of using reduce because it would outline the purpose of this code. i.e. to produce the output: existsUnprofiled.